internal/ecrconsumer/image_values.go

```diff
@@ -0,0 +1,84 @@
+package ecrconsumer
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+type ImageEditor interface {
+	Update() error
+}
+
+type Image struct {
+	service    string
+	Registry   string
+	Repository string
+	Tag        string
+}
+
+func LoadImage(serviceName string, client GitCommands) (*Image, error) {
+	resourceBytes, err := client.DownloadFile("master", filepath.Join("k8s/custom-resources", serviceName+".yaml")) // we assume this file path until a location in miranda for custom resources is decided upon
+	if err != nil {
+		return nil, err
+	}
+
+	customResource := CRYaml{}
+
+	err = yaml.Unmarshal(resourceBytes, &customResource)
+	if err != nil {
+		return nil, err
+	}
+
+	arr := strings.Split(customResource.Spec.Values.Image.Repository, "/")
+
+	var registry string
+	var repository string
+	if len(arr) == 2 {
+		repository = arr[1]
+		registry = arr[0]
+	} else {
+		return nil, fmt.Errorf("repository field is invalid! (lacking either the full registry name or repository name)")
+	}
+
+	return &Image{
+		service:    serviceName,
+		Registry:   registry,
+		Repository: repository,
+		Tag:        customResource.Spec.Values.Image.Tag,
+	}, nil
+}
+
+func (i *Image) Update(client GitCommands) error {
```

Comment:
I recommend to avoid smart models that implement business logic like this `Update` method. The `Image` model should be a dumb & simple data type and higher-level behaviours should be implemented elsewhere, like the ecrconsumer's message handler.

If we decompose this function into the pure/impure parts, then we could imagine writing pseudo-code like this in the ecrconsumer:

```
img := getDesiredImageFromECREvent(event)
rls := getHelmReleaseCRFile(releaseName)
putHelmReleaseCRFile(rls.WithImage(img))
```